client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,287 @@
+
+    let numRounds=0;
+    for (let i=0; i < includedSeats.length; i++) {
+      const score = includedSeats[i].get(scoreProperty);
+      if(Array.isArray(score) && score.length > numRounds)
+        numRounds = score.length;
+    }
+    if(this.get('showAllRounds') && Array.isArray(rounds))
+      numRounds = Math.max(rounds.length, numRounds);
+
+    let pScores = []; // Array of scores. pScores[i][0] is player name or team name, last is total (or last score if showTotals is false)
+    if(Array.isArray(seats)) { // Show individual seats
+      // Fill player score array, totals array
+      for (let i=0; i < seats.length; i++) {
+        const score = seats[i].get(scoreProperty);
+        let total;
+        if(Array.isArray(score)) {
+          pScores[i] = score.map( s => isNaN(parseFloat(s)) ? 0 : parseFloat(s));
+          total = this.getTotal(pScores[i]);
+        } else {
+          pScores[i] = [];
+          total = parseFloat(score);
+          if(isNaN(total))
+            total = 0;
+        }
+        pScores[i] = pScores[i].concat(Array(numRounds).fill('')).slice(0,numRounds);
+        if(showTotals)
+          pScores[i].push(total);
+        pScores[i].unshift(seats[i].get('player'));
+      }
+
+      // Sort player scores as requested
+      if(sortField == 'total' && !showTotals) // Use default sort if no totals
+        sortField = 'index';
+      if(sortField == 'total')
+        pScores.sort((a,b) =>a[a.length-1] < b[b.length-1] ? -1 : 1)
+      else
+        pScores.sort((a,b) => {
+          const pa = includedSeats.filter(x=> x.get('player') == a[0])[0].get(sortField);
+          const pb = includedSeats.filter(x=> x.get('player') == b[0])[0].get(sortField);
+          return pa < pb ? -1 : 1;
+        });
+      if(!this.get('sortAscending'))
+        pScores = pScores.reverse();
+
+      // Get player colors if needed
+      if(usePlayerColors)
+        for (let i=0; i < seats.length; i++)
+          colors.push(seats.filter(x=> x.get('player') == pScores[i][0])[0].get('color'));
+    } else if(typeof seats == 'object') { // Display team scores
+      usePlayerColors = false; // No predefined colors for teams
+      let i = 0;
+      for (const team in seats) {
+        // Get array of (arrays of) seat scores.
+        const seatScores = seats[team].map(w =>  asArray(w.get(scoreProperty)));
+
+        // Make all score arrays for this the same length, then add them element-by-element
+        const n = seatScores.reduce((max, xs) => Math.max(max, xs.length), 0);
+        pScores[i] = Array(n).fill(0).map((_,i) => this.getTotal(seatScores.map(xs => xs[i] || 0)));
+        pScores[i] = pScores[i].concat(Array(numRounds).fill('')).slice(0,numRounds);
+
+        // Add totals and team name
+        if(showTotals)
+          pScores[i].push(this.getTotal(pScores[i]));
+        pScores[i].unshift(team);
+        i++
+      }
+    } else // 'seats' property is not null, array, or object, ignore it.
+      return null;
```

Comment:
This should probably still clear the table.

Reply:
This actually can't happen. I'll arrange to clear the table directly from `applyDeltaToDOM`.